### PR TITLE
"Select its parent tab" setting now works as expected

### PR DESCRIPTION
**Pull request by bsclifton**
"Select its parent tab" setting now works as expected
- Properly get openerTabId; it's located in the frame (not the tab)
- Don't consider if tab was active or not

Fixes https://github.com/brave/browser-laptop/issues/9395

Auditors: @bridiver, @BrendanEich

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by bsclifton** [comment]
Requires https://github.com/brave/muon/pull/214 before this can be merged

**Comment by bsclifton** [comment]
ping @bridiver

File: app/common/state/tabState.js [hunk 1]
```diff
@@ -499,7 +499,15 @@ const tabState = {
   },
 
   getOpenerTabId: (state, tabId) => {
-    return tabState.getTabPropertyByTabId(state, tabId, 'openerTabId', tabState.TAB_ID_NONE)
+    const openerTabId = tabState.getTabPropertyByTabId(state, tabId, 'openerTabId', tabState.TAB_ID_NONE)
+    if (openerTabId !== tabState.TAB_ID_NONE) {
+      // Validate that tabId exists
+      const index = getTabInternalIndexByTabId(state, openerTabId)
```
Comment (bridiver) [inline app/common/state/tabState.js]: I would probably use `getByTabId` here and it already handles TAB_ID_NONE by returning null. `getTabInternalIndex` is just a temporary hack until we can re-organize the tab state so best to limit its usage to the smallest set of methods possible

File: app/browser/reducers/tabsReducer.js [hunk 2]
```diff
@@ -51,15 +51,14 @@ const updateActiveTab = (state, closeTabId) => {
     case tabCloseAction.PARENT:
       {
         const openerTabId = tabState.getOpenerTabId(state, closeTabId)
-        const lastActiveTabId = tabState.getLastActiveTabId(state, windowId)
-        if (openerTabId === lastActiveTabId) {
+        if (openerTabId && openerTabId !== tabState.TAB_ID_NONE) {
```
Comment (bridiver) [inline app/browser/reducers/tabsReducer.js]: is the `null` check here necessary?

**Comment by bsclifton** [comment]
nits fixed 😄  merging

**Pull request merged**